api4/post_test.go

```diff
@@ -565,16 +512,13 @@ func TestCreatePostSendOutOfChannelMentions(t *testing.T) {
 			}
 
 			wpost := model.PostFromJson(strings.NewReader(event.Data["post"].(string)))
-			if acm, ok := wpost.Props[model.PROPS_ADD_CHANNEL_MEMBER].(map[string]interface{}); !ok {
-				t.Fatal("should have received ephemeral post with 'add_channel_member' in props")
-			} else {
-				if acm["post_id"] == nil || acm["user_ids"] == nil || acm["usernames"] == nil {
-					t.Fatal("should not be nil")
-				}
-			}
+
+			acm, ok := wpost.Props[model.PROPS_ADD_CHANNEL_MEMBER].(map[string]interface{})
+			require.True(t, ok, "should have received ephemeral post with 'add_channel_member' in props")
+			require.True(t, acm["post_id"] != nil && acm["user_ids"] != nil && acm["usernames"] != nil, "should not be nil")
```

Comment:
To avoid loosing information on the checks, could we change this to something like:

```suggestion
	require.NotNil(t, acm["post_id"])
	require.NotNil(t, acm["user_ids"])
	require.NotNil(t, acm["usernames"])
```

Reply:
Thanks @mgdelacroix, this case is now separated out into its constituent assertions.